Jenkinsfile

```diff
@@ -0,0 +1,116 @@
+pipeline {
+    agent {
+        ecs {
+            inheritFrom 'terraform'
+        }
+    }
+    environment {
+        stage = getStageFromBranch()
+        account_number = getAccountNumberFromBranch()
+    }
+    stages {
+        stage('Configure AWS credentials') {
+            steps {
+                echo 'Configuring AWS credentials...'
+                sh 'java -version'
+                sh "aws configure set role_arn arn:aws:iam::${env.account_number}:role/${env.stage}-terraform-role --profile ${env.stage}terraform"
+                sh "aws configure set region eu-west-2 --profile ${env.stage}terraform"
+                sh "aws configure set source_profile default --profile ${env.stage}terraform"
+                withCredentials([[
+                                         $class: 'AmazonWebServicesCredentialsBinding',
+                                         credentialsId: "${env.stage}Terraform",
+                                         accessKeyVariable: 'AWS_ACCESS_KEY_ID',
+                                         secretKeyVariable: 'AWS_SECRET_ACCESS_KEY'
+                                 ]]) {
+                    sh 'aws configure set aws_access_key_id ${AWS_ACCESS_KEY_ID}'
+                    sh 'aws configure set aws_secret_access_key ${AWS_SECRET_ACCESS_KEY}'
```

Comment:
Did you look into using an ECS task role rather than an AWS key to do the Terraform authentication? Sam did something similar for the frontend deployment in https://github.com/nationalarchives/tdr-terraform-backend/pull/8

Reply:
I'm hoping to do this, now that Sam has got this working.